agent/dns.go

```diff
@@ -1030,12 +1060,29 @@ func (d *DNSServer) lookupServiceNodes(datacenter, service, tag string, connect
 		QueryOptions: structs.QueryOptions{
 			Token:      d.agent.tokens.UserToken(),
 			AllowStale: d.config.AllowStale,
+			MaxAge:     d.config.CacheMaxAge,
 		},
 	}
 
 	var out structs.IndexedCheckServiceNodes
-	if err := d.agent.RPC("Health.ServiceNodes", &args, &out); err != nil {
-		return structs.IndexedCheckServiceNodes{}, err
+
+	if d.config.UseCache {
```

Comment:
I liked how you broke out the cache get/rpc in the lookupNode function. Could you do the same here for HealthServiceNodes?

Reply:
This func does not do much aside from just making the RPC call, I've left it alone

Reply:
Right you are.